source/data-modeling/associations.txt

```diff
@@ -0,0 +1,504 @@
+.. literalinclude:: /includes/data-modeling/associations.rb
+   :language: ruby
+   :start-after: # start-embed-many
+   :end-before: # end-embed-many
+
+{+odm+} stores documents embedded with the ``embeds_many`` macro in the
+parent document as an array field with the same name as the embedded class. The
+preceding ``Album`` documents are stored in the ``Band`` document as shown in
+the following example:
+
+.. literalinclude:: /includes/data-modeling/associations.rb
+   :language: ruby
+   :start-after: # start-embed-many-stored
+   :end-before: # end-embed-many-stored
+
+You can store the embedded document with a different name by using the
+``store_as`` option, as shown in the following example:
+
+.. literalinclude:: /includes/data-modeling/associations.rb
+   :language: ruby
+   :start-after: # start-embed-many-store-as
+   :end-before: # end-embed-many-store-as
+
+Recursive Embedding
+~~~~~~~~~~~~~~~~~~~
+
+You can recursively embed one or more documents of the same type into a parent
+class by using the ``recursively_embeds_one`` and ``recursively_embeds_many``
+macros. Both macros provide accessors for the parent and child documents through
+a ``parent_*`` method and a ``child_*`` method, where ``*`` represents the name
+of the class. The following example creates a ``Band`` class that recursively
+embeds multiple other ``Band`` documents to represent multiple band names:
+
+.. literalinclude:: /includes/data-modeling/associations.rb
+   :language: ruby
+   :start-after: # start-recursive-embed
+   :end-before: # end-recursive-embed
+
+You can access the parent and child documents through the ``parent_band`` and
+``child_band`` methods, as shown in the following example:
+
+.. literalinclude:: /includes/data-modeling/associations.rb
+   :language: ruby
+   :start-after: # start-recursive-embed-access
+   :end-before: # end-recursive-embed-access
+
+Query Embedded Associations
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can access embedded documents when querying the collection of the parent
+class by using dot notation.
+
+The following example uses dot notation to query ``Tour`` type documents that
+are embedded in a ``Band`` class. The query returns documents that have a
+``tours.year`` value of ``2000`` or greater:
+
+.. literalinclude:: /includes/data-modeling/associations.rb
+   :language: ruby
+   :start-after: # start-embedded-query
+   :end-before: # end-embedded-query
+
+You can use the ``pluck()`` projection method to retrieve embedded documents
+without retrieving their associated parent documents, as shown in the following
+example:
+
+.. literalinclude:: /includes/data-modeling/associations.rb
+   :language: ruby
+   :start-after: # start-embedded-query-pluck
+   :end-before: # end-embedded-query-pluck
+
+You can use {+odm+} query methods to query on embedded associations of documents
+that are already loaded in the application. This is called "embedded matching"
+and is implemented by {+odm+} without sending queries to the server.
```

Comment:
S: not sure if you need the quotes around embedded matching, maybe you could reword to:
```suggestion
You can use {+odm+} query methods to perform embedded matching, or to query on embedded associations of documents
that are already loaded in the application. {+odm+} implements embedded matching without sending queries to the server.
```